src/NBitcoin/IBitcoinSerializable.cs

```diff
@@ -55,9 +68,9 @@ public static int GetSerializedSize(this IBitcoinSerializable serializable, Prot
             return GetSerializedSize(serializable, version, SerializationType.Disk);
         }
 
-        public static void ReadWrite(this IBitcoinSerializable serializable, byte[] bytes, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION, Network network = null)
+        public static void ReadWrite(this IBitcoinSerializable serializable, byte[] bytes, Network network, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION)
```

Comment:
You probably want to create another method that does not need network (and instead use DefaultConsensuFactory).